server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java

```diff
@@ -333,11 +344,22 @@ protected void stop() {
          stop(i);
          log.infof("Stopped container %d", i);
       }
-      ResourceReaper.instance().performCleanup();
+      removeDockerImage(image.getDockerImageName());
       // See https://github.com/testcontainers/testcontainers-java/issues/2276
       ThreadLeakChecker.ignoreThreadsContaining("docker-java-stream-");
    }
 
+   public static void removeDockerImage(String dockerImageName) {
+      try {
+         DockerClient dockerClient = DockerClientFactory.instance().client();
+         dockerClient.removeImageCmd(dockerImageName).withForce(true).exec();
+         log.infof("Removed image %s", dockerImageName);
+         dockerClient.pruneCmd(PruneType.BUILD).exec();
```

Comment:
Maybe we need to prune images as well? I'm not sure what qualifies as "build cache", but I would imagine that since the image build was finished, intermediate layers in the image we just removed may be considered actual images, not part of the build cache.